opm/simulators/wells/BlackoilWellModel_impl.hpp

```diff
@@ -1229,7 +1234,144 @@ namespace Opm {
         return {more_network_update, well_group_control_changed};
     }
 
+    // This function is to be used for well groups in an extended network that act as a subsea manifold
+    // The wells of such group should have a common THP and total phase rate(s) obeying (if possible) 
+    // the well group constraint set by GCONPROD
+    template <typename TypeTag>
+    void
+    BlackoilWellModel<TypeTag>::
+    computeWellGroupThp(const double dt, DeferredLogger& local_deferredLogger)
+    {
+        const int reportStepIdx = this->ebosSimulator_.episodeIndex();
+        const auto& network = schedule()[reportStepIdx].network();
+
+        if (!network.active()) {
+            return;
+        }
+
+        auto& well_state = this->wellState();
+        auto& group_state = this->groupState();
+
+        for (const std::string& nodeName : network.node_names()) {
+            const bool has_choke = network.node(nodeName).as_choke();
+            if (has_choke) {
+                const auto& summary_state = this->ebosSimulator_.vanguard().summaryState();
+                const Group& group = schedule().getGroup(nodeName, reportStepIdx);
+                std::optional<Group::ProductionControls> ctrl = group.productionControls(summary_state);
+                const auto cmode = ctrl->cmode;
+                const auto pu = this->phase_usage_;
+                std::vector<double> resv_coeff(pu.num_phases, 1.0);
+                double gratTargetFromSales = 0.0; 
```

Comment:
This is something I am not familiar, can we do the following instead of keeping a 0 there?

```c++
451     if (group_state.has_grat_sales_target(group.name())) {
452         gratTargetFromSales = group_state.grat_sales_target(group.name());
          }
 
```

Reply:
I think we can. So I put it there (but did not test it yet).